react-native/app/index.tsx

```diff
@@ -9,38 +10,57 @@ import {
   View,
 } from "react-native";
 import { createClient, Code, ConnectError } from "@connectrpc/connect";
-import { createXHRGrpcWebTransport } from "./custom-transport";
+import { createConnectTransport } from "@connectrpc/connect-web";
 import {
   ElizaService,
   IntroduceRequestSchema,
 } from "../gen/connectrpc/eliza/v1/eliza_pb.js";
-// Needed to polyfill TextEncoder/ TextDecoder
-import "fast-text-encoding";
 import { create } from "@bufbuild/protobuf";
-import { polyfills } from "./polyfills";
 
-polyfills();
-
-interface Response {
+interface ConvoResponse {
   text: string;
   sender: "eliza" | "user";
 }
 
 function Index() {
   const [statement, setStatement] = useState<string>("");
   const [introFinished, setIntroFinished] = useState<boolean>(false);
-  const [responses, setResponses] = useState<Response[]>([
+  const [responses, setResponses] = useState<ConvoResponse[]>([
     {
       text: "What is your name?",
       sender: "eliza",
     },
   ]);
 
-  // Make the Eliza Service client
+  const abort = new AbortController();
+
+  // Make the Eliza Service client using grpc-web transport.
   const client = createClient(
     ElizaService,
-    createXHRGrpcWebTransport({
+    // createGrpcWebTransport({
+    createConnectTransport({
       baseUrl: "https://demo.connectrpc.com",
+      // Customize fetch with the the Expo fetch implementation
+      fetch: (input, init) => {
+        return fetch(input.toString(), {
+          ...init,
+          body: init?.body ?? undefined,
+          credentials: init?.credentials ?? undefined,
+          signal: init?.signal ?? undefined,
+        });
+        // TODO - Need to return a type of Response here instead of Expo's FetchResponse
+        // otherwise it fails compilation
+        // return new Promise<Response>((resolve) => {
+        //   fetch(input.toString(), {
+        //     ...init,
+        //     body: init?.body ?? undefined,
+        //     credentials: init?.credentials ?? undefined,
+        //     signal: init?.signal ?? undefined,
+        //   }).then((resp) => {
+        //     resolve(new Response(resp.body));
+        //   });
+        // });
```

Comment:
expo/fetch brings it's own types, even though expo/tsconfig.base includes the DOM lib. Little we can do except cast (we should actually test for input to be a string though).


```suggestion
      // Customize fetch with the Expo fetch implementation
      fetch: (input, init) => {
        if (typeof input !== "string") {
          throw new Error("expo/fetch requires the first argument to be a string URL");
        }
        return fetch(input, {
          ...init,
          body: init?.body ?? undefined,
          credentials: init?.credentials ?? undefined,
          signal: init?.signal ?? undefined,
        }) as unknown as Promise<Response>;
```